R/filter_confirmation.R

```diff
@@ -0,0 +1,233 @@
+#' Filter Confirmed Assessments
+#'
+#' The function filters observation using a condition taking subsequent
+#' observations into account. For example, it could select all observations with
+#' `AVALC == "Y"` and `AVALC == "Y"` for at least one subsequent observation.
+#' The input dataset is joined with itself to enable conditions taking variables
+#' from both the current observation and the subsequent observations into
+#' account. The suffix ".join" is added to the variables from the subsequent
+#' observations.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified for `by_vars`, `join_vars`, and `order` are
+#'   expected.
+#'
+#' @param by_vars By variables
+#'
+#'   The specified variables are used as by variables for joining the input
+#'   dataset with itself.
+#'
+#' @param join_vars Variables to keep from joined dataset
+#'
+#'   The variables needed from the subsequent observations should be specified
+#'   for this parameter. The specified variables are added to the joined dataset
+#'   with suffix ".join". For example to select all observations with `AVALC ==
+#'   "Y"` and `AVALC == "Y"` for at least one subsequent visit `join_vars =
+#'   vars(AVALC, AVISITN)` and `filter = AVALC == "Y" & AVALC.join == "Y" &
+#'   AVISITN < AVISITN.join` could be specified.
+#'
+#' @param first_cond Condition for selecting range of data
+#'
+#'   If this argument is specified, the subsequent observations are restricted
+#'   up to the first observation where the specified condition is fulfilled. If
+#'   the condition is not fulfilled for any of the subsequent observations, all
+#'   observations are removed.
+#'
+#' @param order Order
+#'
+#'   The observations are ordered by the specified order.
+#'
+#' @param filter Condition for selecting observations
+#'
+#'   The filter is applied to the joined dataset for selecting the confirmed
+#'   observations. The condition can include summary functions. The joined
+#'   dataset is grouped by the original observations. I.e., the summary function
+#'   are applied to all observations up to the confirmation observation. For
+#'   example, `filter = AVALC == "CR" & all(AVALC.join %in% c("CR", "NE")) &
+#'   count_vals(var = AVALC.join, val = "NE") <= 1` selects observations with
+#'   response "CR" and for all observations up to the confirmation observation
+#'   the response is "CR" or "NE" and there is at most one "NE".
+#'
+#' @param check_type Check uniqueness?
+#'
+#'   If `"warning"` or `"error"` is specified, the specified message is issued
+#'   if the observations of the input dataset are not unique with respect to the
+#'   by variables and the order.
+#'
+#'   *Default:* `"none"`
+#'
+#'   *Permitted Values:* `"none"`, `"warning"`, `"error"`
+#'
+#' @author Stefan Bundfuss
+#'
+#' @keywords adam user_utility
+#'
+#' @export
+#'
+#' @examples
+#'
+#' library(dplyr)
+#' library(admiralonco)
```

Comment:
apologies if stupid question but do we not need to load admiral too here?

Reply:
You need to load admiral only if any admiral function is called directly in the example. You do not need to load it if the admiral functions are called within the admiralonco function only.

Reply:
ah i see, thanks